src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -217,6 +227,41 @@ public function registerAliasesForSinglyImplementedInterfaces(): void
         $this->interfaces = $this->singlyImplemented = $this->aliases = [];
     }
 
+    protected function loadExtensionConfig(string $namespace, array $config): void
+    {
+        if (!$this->prepend) {
+            $this->container->loadFromExtension($namespace, $config);
+
+            return;
+        }
+
+        if ($this->importing) {
+            if (!isset($this->extensionConfigs[$namespace])) {
+                $this->extensionConfigs[$namespace] = [];
+            }
+            array_unshift($this->extensionConfigs[$namespace], $config);
+
+            return;
+        }
+
+        $this->container->prependExtensionConfig($namespace, $config);
+    }
+
+    protected function loadExtensionConfigs(): void
```

Comment:
do we have to make those 2 methods protected?

Reply:
I see that child classes call them. What about making them final?

Reply:
Done Nicolas! 🙏